src/main/java/org/junit/runners/Parameterized.java

```diff
@@ -365,7 +396,13 @@ private TestWithParameters createTestWithNotNormalizedParameters(
         @SuppressWarnings("unchecked")
         private static List<Object> allParameters(
                 TestClass testClass, FrameworkMethod parametersMethod) throws Throwable {
-            Object parameters = parametersMethod.invokeExplosively(null);
+            Object parameters;
+            try {
+                parameters = parametersMethod.invokeExplosively(null);
+            } catch (AssumptionViolatedException e) {
```

Comment:
Couldn't you catch this exception in the caller (`RunnersFactory`)? If the exception gets caught, you can set a field in `RunnersFactory` named `runnerOverride` to a new `AssumptionViolationRunner`. If `runnerOverride` is set it would be returned from `RunnersFactory.createRunners()`.

As a bonus, we could possibly catch any exception from there, and for the exceptions that aren't `AssumptionViolatedException ` use `ErrorReportingRunner`.